eskip/string.go

```diff
@@ -99,8 +100,18 @@ func (r *Route) predicateString(sortPredicates bool) string {
 		predicates = appendFmtEscape(predicates, `Method("%s")`, `"`, r.Method)
 	}
 
-	for k, v := range r.Headers {
-		predicates = appendFmtEscape(predicates, `Header("%s", "%s")`, `"`, k, v)
+	headerKeys := make([]string, 0, len(r.Headers))
+
+	for k := range r.Headers {
+		headerKeys = append(headerKeys, k)
+	}
+
+	if sortHeaderPredicateValues {
```

Comment:
The list is small and sorting would be super fast, the main issue is allocations to copy names.

We should either drop the flag and always sort of create two branches such that when sorting is disabled we avoid copying header names.

Reply:
> We should either drop the flag and always sort of create two branches

Since we need them sorted only for a specific purpose lets have the flag and two branches

Reply:
> We should either drop the flag and always sort of create two branches

Since we need them sorted only for a specific purpose lets have the flag and two branches